src/app/modules/visualizacion-informacion/components/bi-queries/bi-queries.component.ts

```diff
@@ -7,9 +14,108 @@ import { Component, OnInit } from '@angular/core';
 
-  constructor() { }
+  public consultaSeleccionada;
+  public opciones = [];
+  public queries$: Observable<any>;
+  public organizaciones$: Observable<any>;
+  public argumentos;
+  public argumentosSalida = [];
+  public resultados;
+  public mostrarSalida = false;
+  public tipoPrestaciones;
+
+  constructor(private queryService: QueriesService,
+    private profesionalService: ProfesionalService,
+    private auth: Auth,
+    private router: Router,
+    public servicioPrestacion: TipoPrestacionService
+  ) { }
 
   ngOnInit() {
+    const permisos = this.auth.getPermissions('visualizacionInformacion:biQueries:?');
+    if (permisos.length) {
+      if (permisos[0] === '*') {
+        this.queries$ = this.queryService.getAllQueries({ desdeAndes: true });
+      } else {
+        this.queries$ = this.queryService.getAllQueries({ _id: permisos });
+      }
+    } else {
+      this.router.navigate(['./inicio']);
+    }
+
+  }
+
+  getArgumentos() {
+    if (this.consultaSeleccionada) {
+      this.argumentos = this.consultaSeleccionada.argumentos;
+      this.organizaciones$ = this.auth.organizaciones();
+    }
+  }
+
+  loadProfesionales(event) {
+    let listaProfesionales = [];
+    if (event.query) {
+      let query = {
+        nombreCompleto: event.query
+      };
+      this.profesionalService.get(query).subscribe(resultado => {
+        listaProfesionales = resultado;
+        event.callback(listaProfesionales);
+      });
+    } else {
+      event.callback(listaProfesionales);
+    }
   }
 
+  loadConceptosTurneables(event) {
+    this.servicioPrestacion.get({ turneable: 1 }).subscribe(event.callback);
+  }
+
+  loadEstadosAgenda(event) {
+    const estadosAgendas = [
+      { id: 'planificacion', nombre: 'Planificación' },
+      { id: 'disponible', nombre: 'Disponible' },
+      { id: 'publicada', nombre: 'Publicada' },
+      { id: 'suspendida', nombre: 'Suspendida' },
+      { id: 'pausada', nombre: 'Pausada' },
+      { id: 'pendienteAsistencia', nombre: 'Pendiente Asistencia' },
+      { id: 'pendienteAuditoria', nombre: 'Pendiente Auditoría' },
+      { id: 'auditada', nombre: 'Auditada' },
+      { id: 'borrada', nombre: 'Borrada' }
+    ];
+    event.callback(estadosAgendas);
+  }
+
+  loadEstadosTurno(event) {
+    const estadosTurnos = [
+      { id: 'disponible', nombre: 'Disponible' },
+      { id: 'asignado', nombre: 'Asignado' },
+      { id: 'suspendido', nombre: 'Suspendido' },
+      { id: 'turnoDoble', nombre: 'Turno Doble' }
+    ];
+    event.callback(estadosTurnos);
+  }
+
+  descargar() {
+    let resultado = this.queries$.find(query => query.nombre === this.consultaSeleccionada.nombre);
+    if (resultado) {
+      const params = {};
+
+      this.argumentos.forEach(arg => {
+        const key = arg.key;
+        const valor = this.argumentos[key];
+        params[key] = valor;
+        if (valor instanceof Date) {
+          params[key] = moment(valor).format();
```

Comment:
El formateo lo hace el Server de @andes/shared automaticamente. 
```suggestion
          params[key] = valor;
```

Reply:
Hecho